mayavi/plugins/envisage_engine.py

```diff
@@ -31,7 +31,7 @@ class EnvisageEngine(Engine):
     __version__ = 0
 
     # The envisage application.
-    window = Instance(WorkbenchWindow)
+    application = Instance(TasksApplication)
```

Comment:
I think that you probably want the Envisage `TaskWindow` here rather than the `TasksApplication`, going by straight analogy.  The reason is that a `TasksApplication` can have multiple `TaskWindows` and so some of the events you are listening to can fire when you don't expect them to.

Note that there is a minor bug here in that the old code: the `window` object is assuming an Envisage `WorkbenchWindow` and not a Pyface one (eg. `get_service` does not exist on the Pyface version).

Reply:
Addendum: or if this `EnvisageEngine` is global state for the app rather than a single window, then you need to adjust the event you are listening to for to be things like "application stop", etc. instead of "window close".